test/Microsoft.TestPlatform.AcceptanceTests/AppDomainTests.cs

```diff
@@ -24,28 +25,39 @@ public void RunTestExecutionWithDisableAppDomain(RunnerInfo runnerInfo)
         {
             AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerInfo);
 
-            var testResults = GetResultsDirectory();
-            var testAppDomainDetailFileName = Path.Combine(Path.GetTempPath(), "appdomain_test.txt");
-            var dataCollectorAppDomainDetailFileName = Path.Combine(Path.GetTempPath(), "appdomain_datacollector.txt");
+            using var workspace = new Workspace();
+            var testAppDomainDetailFileName = Path.Combine(workspace.Path, "appdomain_test.txt");
+            var dataCollectorAppDomainDetailFileName = Path.Combine(workspace.Path, "appdomain_datacollector.txt");
 
             // Delete test output files if already exist
             File.Delete(testAppDomainDetailFileName);
             File.Delete(dataCollectorAppDomainDetailFileName);
+
             var runsettingsFilePath = this.GetInProcDataCollectionRunsettingsFile(true);
             var arguments = PrepareArguments(
                 this.GetSampleTestAssembly(),
                 this.GetTestAdapterPath(),
                 runsettingsFilePath,
                 this.FrameworkArgValue,
                 runnerInfo.InIsolationValue,
-                testResults);
+                workspace.Path);
+
+            // Sets the environment variables used by the test project and test data collector.
+            var env = new Dictionary<string, string>
```

Comment:
This looks fine to me. Normal ProcessStartInfo also uses dictionary.

Reply:
I was more talking about the magic env var string not the dictionary per say (i.e. instead of having the various places using "AppDomainTestFilePath" we could have a class with a static/const string).

Reply:
I'd say duplicate the code. Tests should not be DRY.